src/UbidotsFONA.cpp

```diff
@@ -0,0 +1,521 @@
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSTART"));
+#endif
+        return false;
+    }
+    fonaSS.print(F("AT+CIPSEND="));
+    fonaSS.println(dataLen(allData));
+    if (strstr(readData(4000), ">") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSEND"));
+#endif
+        return false;
+    }
+    fonaSS.println(allData);
+    char* resp = readData(4000);
+    response = strtok(resp, "|");
+    serverResponse = response;
+    while (response!=NULL) {
+        printf("%s\n", response);
+        response = strtok(NULL, "|");
+        if (response != NULL) {
+            serverResponse = response;
+        }
+    }
+    num = atof(serverResponse);
+
+    fonaSS.println(F("AT+CIPCLOSE"));
+    if (strstr(readData(4000), "CLOSE OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error sending data"));
+#endif
+        return false;
+    }
+    fonaSS.println(F("AT+CIPSHUT"));
+    if (strstr(readData(4000), "SHUT OK") == NULL) {
+#ifdef DEBUG_UBIDOTS
+        Serial.println(F("Error with AT+CIPSHUT"));
+#endif
+        return false;
+    }
+    free(allData);
+    return num;
+}
+
+int Ubidots::readLine(uint32_t ts_max, bool multiline) {
+    int replyidx = 0;
+    while (ts_max--) {
+        if (replyidx >= 254) {
+            break;
+        }
+        while (fonaSS.available()) {
+            char c =  fonaSS.read();
+            if (c == '\r') continue;
+            if (c == 0xA) {
+            if (replyidx == 0)   // the first 0x0A is ignored
+                continue;
+            }
+            if (!multiline) {
+                ts_max = 0;         // the second 0x0A is the end of the line
+                break;
+            }
+            buffer[replyidx] = c;
+            replyidx++;
+        }
+        if (ts_max == 0) {
+            break;
+        }
+        delay(1);
+    }
+    buffer[replyidx] = 0;  // null term
+    Serial.println(buffer);
+    return replyidx;
+}
+
+
+/***************************************************************************
+AUXILIAR FUNCTIONS
+***************************************************************************/
+
+/**
+ * Assigns a new device name
+ * @arg device_name new name that you want to assign to your device
+ */
+void Ubidots::setDeviceName(char* device_name) {
+    _device_name = device_name;
+}
+
+/**
+ * Assigns a new device label
+ * @arg device_label new label that you want to assign to your device
+ */
+void Ubidots::setDeviceLabel(char* device_label) {
+    _device_label = device_label;
+}
+
+/**
+ * Gets the length of a variable
+ * @arg variable a variable of type char
+ * @return dataLen the length of the variable
+ */
+int Ubidots::dataLen(char* variable) {
+  uint8_t dataLen = 0;
+  for (int i = 0; i <= 250; i++) {
+    if (variable[i] != '\0') {
+      dataLen++;
+    } else {
+      break;
+    }
+  }
+  return dataLen;
+}
+
+
+// ------------------------------------------------------------------
```

Comment:
If these functions were deprecated, do not add them. We grow up the number version to 3.0 so we should not include any deprecated function